### PR TITLE
Add Cloudflare preview action

### PR DESCRIPTION
**Pull request by dweng0**
This PR adds an action that will allow PRs to be provided with preview links

### CODE REVIEW

File: .github/workflows/cloudflare-preview.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+on:
+  pull_request:
+    branches:
+      - develop
+name: Cloudflare-preview
+jobs:
+  build-deploy:
```
Comment (aiman) [inline .github/workflows/cloudflare-preview.yml]: Please choose a unique job name, as this name is already used in this repo. I would suggest `cloudflare-preview`.

File: .github/workflows/cloudflare-preview.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        uses: actions/checkout@v4
+
+      - name: Set up Quarto
+        uses: quarto-dev/quarto-actions/setup@v2
+
+      - name: Render and Publish
+        uses: quarto-dev/quarto-actions/render@v2
+        with:
+          to: html
+      - name: Publish to Cloudflare Pages
```
Comment (aiman) [inline .github/workflows/cloudflare-preview.yml]: Is there a reason there are spaces in between previous steps but not here?

```suggestion
          to: html

      - name: Publish to Cloudflare Pages
```

File: .github/workflows/cloudflare-preview.yml (hunk 1 above)
Comment (aiman) [inline .github/workflows/cloudflare-preview.yml]: e.g.
```suggestion
  cloudflare-preview:
```

Comment (dweng0) [reply .github/workflows/cloudflare-preview.yml]: have updated the yaml file to use the name `cloudflare-deploy`

File: .github/workflows/cloudflare-preview.yml (hunk 2 above)
Comment (dweng0) [reply .github/workflows/cloudflare-preview.yml]: Have updated this to have spaces consistent across the steps for this file.

**Comment by aiman** [comment]
Thanks!

**Pull request merged**